src/plan/aggregate.rs

```diff
@@ -77,10 +98,10 @@ impl<P: Implementable> Implementable for Aggregate<P> {
                         let min = vals[0].0;
                         output.push((*min, 1));
                     })
-                    .map(|(key, min)| {
+                    .map(move |(key, min)| {
                         let mut v = key.clone();
                         v.push(Value::Number(min as i64));
-                        v
+                        reorder(v, sym_position.clone())
```

Comment:
The reorder stuff seems a bit suspcious to me. 

1. Why do we suddenly need to re-order anyways? (I'm sure there is a good reason, I just didn't get it)
2. Why can we not insert the result directly at the correct offset rather than tacking it on at the end and sorting? 

Another thing is that `sym_position` is copied on every invocation of the closure (not sure whether the compiler is smart enough to figure this out). I think nothing prevents us from passing positions by reference.

If `reorder` proves to be absolutely neccessary, then we should consider adding an `#[inline(always)]` annotation.

Reply:
We did simply append the result from the aggregations into the key vector. But what if the query looks like that: `:find ?e (count ?age) ?u` or any variation of it. 
I suppose I could change the logic to insert at the correct position.